### PR TITLE
Naptan batch getter

### PR DESCRIPTION
**Pull request by Danielleoptibus**

### CODE REVIEW

File: naptan_batch_getter/naptan_get_batch_stops.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import boto3
+from botocore.exceptions import ClientError
+
+TABLE_NAME = 'Naptan'
+MAX_BARCH_SIZE = 100
```
Comment (urishalit) [inline naptan_batch_getter/naptan_get_batch_stops.py]: BARCH or BATCH? :cat:

File: naptan_batch_getter/naptan_get_batch_stops.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+def lambda_handler(event, context):
+
+    # Connect to DynamoDB using boto
+    dynamo_db = boto3.resource('dynamodb', region_name=REGION)
+    stop_ids = event["stop_ids"]
+    stops_ids_list = stop_ids.split(',')
+
+    if len(stops_ids_list) > MAX_BARCH_SIZE:
+        print('Cannot support batch get item for more then 100 items')
+        raise ClientError
```
Comment (urishalit) [inline naptan_batch_getter/naptan_get_batch_stops.py]: I think the raise should also include the error message so the client knows why it failed.

File: naptan_batch_getter/naptan_get_batch_stops.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+def lambda_handler(event, context):
+
+    # Connect to DynamoDB using boto
+    dynamo_db = boto3.resource('dynamodb', region_name=REGION)
+    stop_ids = event["stop_ids"]
+    stops_ids_list = stop_ids.split(',')
+
+    if len(stops_ids_list) > MAX_BARCH_SIZE:
+        print('Cannot support batch get item for more then 100 items')
```
Comment (urishalit) [inline naptan_batch_getter/naptan_get_batch_stops.py]: Max batch size in 100 (batch given is of size %s)

File: naptan_create/create_table.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    # read temp file and add each item to dynamodb
+    with table.batch_writer() as batch:
+        try:
+            # open using codecs for the errors='ignore' flag.
+            with codecs.open("/tmp/stops_temp_file.csv", 'r', encoding='ascii', errors='ignore') as stops_csv:
+                reader = csv.DictReader(stops_csv)
+                for idx, row in enumerate(reader):
+                    # by default, convert empty strings to null
+                    stop = create_stop(row)
+                    batch.put_item(Item=stop)
```
Comment (urishalit) [inline naptan_create/create_table.py]: Let's either change it to use `BatchWriteItem` or add it as a subtask to the next story (Improved...)

File: naptan_create/create_table.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    # download and unzip file, save Stops.csv in tmp file
+    try:
+        with ZipFile(BytesIO(url.read())) as zip_file:
+            for contained_file in zip_file.namelist():
+                with open("/tmp/stops_temp_file.csv", "wb") as output:
+                    if contained_file == 'Stops.csv':
+                        for line in zip_file.open(contained_file).readlines():
+                            output.write(line)
+                        break
+    except Exception:
```
Comment (urishalit) [inline naptan_create/create_table.py]: This is too broad an exception, and the print doesn't give us much value - Let's remove the `try/except`

(Same for the next `try/except` block)

File: natpan_getter/naptan_get_stops.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+    stop_id = event["stopid"]
+
+    try:
+        response = table.get_item(
+            Key={
+                'StopId': stop_id
+            }
+        )
+    except ClientError as e:
```
Comment (urishalit) [inline natpan_getter/naptan_get_stops.py]: Don't we want to reraise?

**Review by urishalit — APPROVED** [review]

File: naptan_batch_getter/naptan_get_batch_stops.py (hunk 1 above)
Comment (Danielleoptibus) [reply naptan_batch_getter/naptan_get_batch_stops.py]: done

File: naptan_batch_getter/naptan_get_batch_stops.py (hunk 2 above)
Comment (Danielleoptibus) [reply naptan_batch_getter/naptan_get_batch_stops.py]: done

File: naptan_batch_getter/naptan_get_batch_stops.py (hunk 3 above)
Comment (Danielleoptibus) [reply naptan_batch_getter/naptan_get_batch_stops.py]: done

File: naptan_create/create_table.py (hunk 5 above)
Comment (Danielleoptibus) [reply naptan_create/create_table.py]: done

File: natpan_getter/naptan_get_stops.py (hunk 6 above)
Comment (Danielleoptibus) [reply natpan_getter/naptan_get_stops.py]: done

File: naptan_create/create_table.py (hunk 4 above)
Comment (Danielleoptibus) [reply naptan_create/create_table.py]: done

**Pull request merged**